src/services/TagsService.js

```diff
@@ -144,12 +145,23 @@ export default class TagsService {
   /**
    * Whether give string compare keyword.
    *
-   * @param {string} str - String to compare with keyword
+   * @param {Array | string} str - String or array of strings to compare with keyword
    *
    * @return {boolean}
    */
   isContainsKeyword(str) {
```

Comment:
Can we add different method for arrays ? I know it will be work, but if we have only one type on enter and exit it better to understand and more flexible in future. May we should rename current method to `isStringContainsKeyword` and add new `isArrayContainsKeyword`. What do you think ?

Reply:
BTW, ```isContainsKeyword``` - is poor english. 

From lexical point of view it will be more correct to name just ```containsKeyword```.
From my point of view as developer  ```isArrayContainsKeyword``` is also a bad name, because it is ambigous - does not explain, how it actually behaves. Because there will be great difference between ```containsAnyKeyword(Array keywords)``` and ```containsAllKeywords(Array keywords)```

Reply:
In my opinion, it's regular when checking a param is array or not. With PHP, we usually check array in query builder, request, reading config...

Reply:
@sariluong As I said, problem is not only in name, but in ambiguous behavior. Ex.
```javascript
this.addTag('tag1');
this.addTag(['tag2', 'tag3', 'tag4']);
```
would have predictable effect - all tags will be added. Here we can use [Ad hoc polymorphism](https://en.wikipedia.org/wiki/Ad_hoc_polymorphism) to accept both strings and arrays, and consider our code clean and readable.

but
```javascript
this.hasTag(['tag2', 'tag3', 'tag4']);
this.containsKeyword(['kw1', 'kw2', 'kw3']);
```
is not clear enough - will it return true when all tags/keywords are present or at least one of them presents? In this case I would consider it non-obvious code design/naming. Yes, you can go and read documentation, but much better, when for array you have dedicated function with self-explanatory name, like 
```javascript
this.containsAllKeywords(['kw1', 'kw2', 'kw3']);
this.containsAnyKeyword(['kw1', 'kw2', 'kw3']);
```
Reading this code first time you don't have to guess, what will happen or dig deeper into code or docs.